api/test/e2e/impact/impact.spec.ts

```diff
@@ -1405,4 +1499,95 @@ describe('Impact Table and Charts test suite (e2e)', () => {
       );
     });
   });
+
+  describe('Filters Tests', () => {
+    test('When I query the API for a Impact table with Location Type filters Then I should get the correct data', async () => {
+      const adminRegion: AdminRegion = await createAdminRegion({
+        name: 'Fake AdminRegion',
+      });
+
+      const unit: Unit = await createUnit({ shortName: 'fakeUnit' });
+      const indicator: Indicator = await createIndicator({
+        name: 'Fake Indicator',
+        unit,
+      });
+
+      const material: Material = await createMaterial({
+        name: 'Fake Material',
+      });
+
+      const businessUnit1: BusinessUnit = await createBusinessUnit({
+        name: 'Fake BusinessUnit 1',
+      });
+
+      const supplier1: Supplier = await createSupplier({
+        name: 'Fake Supplier 1',
+      });
+
+      const sourcingLocation1: SourcingLocation = await createSourcingLocation({
+        material: material,
+        businessUnit: businessUnit1,
+        t1Supplier: supplier1,
+        adminRegion: adminRegion,
+        locationType: LOCATION_TYPES.COUNTRY_OF_PRODUCTION,
+      });
+
+      const sourcingLocation2: SourcingLocation = await createSourcingLocation({
+        material: material,
+        businessUnit: businessUnit1,
+        t1Supplier: supplier1,
+        adminRegion: adminRegion,
+        locationType: LOCATION_TYPES.AGGREGATION_POINT,
+      });
+
+      for await (const [index, year] of [2010, 2011, 2012].entries()) {
+        const startTonnage: number = 100;
+        const indicatorRecord: IndicatorRecord = await createIndicatorRecord({
+          value: 100 + index / 0.02,
+          indicator,
+        });
+
+        await createSourcingRecord({
+          tonnage: startTonnage + 100 * index,
+          year,
+          indicatorRecords: [indicatorRecord],
+          sourcingLocation: sourcingLocation1,
+        });
+      }
+
+      for await (const [index, year] of [2010, 2011, 2012].entries()) {
+        const startTonnage: number = 1000;
+        const indicatorRecord: IndicatorRecord = await createIndicatorRecord({
+          value: 300 + index / 0.02,
+          indicator,
+        });
+
+        await createSourcingRecord({
+          tonnage: startTonnage + 100 * index,
+          year,
+          indicatorRecords: [indicatorRecord],
+          sourcingLocation: sourcingLocation2,
+        });
+      }
```

Comment:
Since the loop goes through the same years, maybe it's worth grouping both indicator and sourcing records into a single for loop, putting in some comments to the code more clear
`for await.....
    //Create Indicator and sourcing Record for country of production Sourcing Location
const startTonnage.....`